xwiki-platform-core/xwiki-platform-panels/xwiki-platform-panels-api/src/main/java/org/xwiki/panels/internal/AbstractPanelsUIExtensionManager.java

```diff
@@ -109,7 +110,18 @@
                     for (UIExtension extension : allExtensions) {
                         // Resolve and serialize the extension id to make it absolute before we can compare it with
                         // the serialized panel reference.
-                        String extensionId = serializer.serialize(resolver.resolve(extension.getId()));
```

Comment:
Not your fault but this code is pretty bad. Going trough serialized references does not make much sense, panelSerializedReferences should be a list of DocumentReference and `extensionId.equals(panelSerializedReference)` should be replaced by an equals between two references.

Reply:
Actually actually panelSerializedReferences should even be replaced with a Set of DocumentReference so that the following code can just use #contains instead of testing each extension with each configured panel.

Reply:
I will look for some refactoring.